### ISSUE TITLE
containers missing in nf-core download

### ISSUE THREAD

**Issue by egenomics** [nf-core/sarek#748]
### Description of the bug

Hi,

I have downloaded sarek latest version using nf-core tools, with all singularity containers (in theory). When I move the pipeline to the cluster (no internet access) and execute it selecting deepvariant as a variant caller it fails locating the deepvariant container and therefore the job fails (no internet).

I can download it manually with the command line 
singularity pull  --name google-deepvariant-1.3.0.img.pulling.1663095569629 docker://google/deepvariant:1.3.0 > /dev/null

but perhaps this is happening for other containers? It might be worth taking a look into the list.

Best,
JL

### Command used and terminal output

```console
Error executing process > 'NFCORE_SAREK:SAREK:GERMLINE_VARIANT_CALLING:RUN_DEEPVARIANT:DEEPVARIANT (WTCHG_722292)'

Caused by:
  Failed to pull singularity image
  command: singularity pull  --name google-deepvariant-1.3.0.img.pulling.1663095569629 docker://google/deepvariant:1.3.0 > /dev/null
  status : 255
  message:
    FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://google/deepvariant:1.3.0: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp 54.242.59.189:443: i/o timeout
```

**Comment by FriederikeHanssen** [comment]
@mirpedrol /anyone else from tools, any idea?

**Comment by egenomics** [comment]
In the meantime, are there instructions somewhere to download the missing containers manually? I can only run the pipeline in the cluster without internet (local machines have very low resources).

If it helps the containers downloaded appear to come from depot.galaxyproject.org, perhaps the other sources are not working?

**Comment by FriederikeHanssen** [comment]
yes, please check here: https://nf-co.re/tools/#downloading-pipelines-for-offline-use

**Comment by egenomics** [comment]
Hi,
I fail to see which are the steps in the manual. It suggests to download the containers using nf-core tools but that is what is failing in the first place (some of the containers are not downloaded).
What I would need to be able to obtain the containers offline is a list of them and which are the commands to apply to them after "singularity pull  --name google-deepvariant-1.3.0.img.pulling.1663095569629 docker://google/deepvariant:1.3.0 > /dev/null"

This is the list of the ones that nf-core tools downloads:
![Screenshot from 2022-09-29 15-38-35](https://user-images.githubusercontent.com/3185205/193061610-fc4bf358-63f2-4510-a70d-8b8004a3bc92.png)

deepvariant and gatk-4.2.6.1 were downloaded manually but when sarek is executed it fails to recognize them and tries (and fails) to download them again as the HPC does not have internet connection.

JL

**Comment by maxulysse** [comment]
Can you try renaming the gatk container to `depot.galaxyproject.org-singularity-gatk4-4.2.6.1--hdfd78af_0.img`.
Not sure what is happening with the deepvariant one, I'll have a look at it.

**Comment by maxulysse** [comment]
`google-deepvariant-1.3.0.img` should actually be the right name for it, can you share any information related to issues for that process?

**Comment by egenomics** [comment]
Hi, basically it does not recognize the container for deepvariant. Based on the size of the files (if you look at the gatk container with the wrong name (manually downloaded) and the one with the right name (the one you pointed out, that was downloaded by nf-core), the size is quite different. Perhaps there are a few commands to configure the containers after the download? 

The exact error is basically that it does not have internet connection:

Error executing process > 'NFCORE_SAREK:SAREK:GERMLINE_VARIANT_CALLING:RUN_DEEPVARIANT:DEEPVARIANT (WTCHG_722292)'

Caused by:
  Failed to pull singularity image
  command: singularity pull  --name google-deepvariant-1.3.0.img.pulling.1664480531548 docker://google/deepvariant:1.3.0 > /dev/null
  status : 255
  message:
    FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://google/deepvariant:1.3.0: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp 3.216.34.172:443: i/o timeout

The actual command used in my HPC script is:
# Define variables for the job analysis
sample=WTCHG_722292
analysis_folder=/well/buck/users/xhs232/analysis/NA12878_OGC_germline/2022_09_29_165536_WTCHG_722292
scripts_folder=$analysis_folder/scripts
genome="GATK.GRCh38"
#bedfile=%BEDFILE%

# Export Singularity variables
export NXF_SINGULARITY_CACHEDIR="/well/buck/users/xhs232/soft/containers"
export SINGULARITY_CACHEDIR="/well/buck/users/xhs232/soft/containers"
module load Nextflow/22.04.0

echo "Running Sarek with sample: "$sample
input=$scripts_folder/samples.csv
nextflow run /well/buck/users/xhs232/soft/nf-core-sarek-3.0.2/workflow -profile singularity --tools haplotypecaller,deepvariant \
--input $input \
--save_output_as_bam \
--genome $genome \
--outdir ${analysis_folder} \
--max_cpus 24

**Comment by maxulysse** [comment]
Can you check if current dev it's solving the issue for you?
If so we'll probably be doing a minor release for it, otherwise we keep digging

**Comment by egenomics** [comment]
Hi,
I just got the results from the dev version and the error is still the same>
Error executing process > 'NFCORE_SAREK:SAREK:GERMLINE_VARIANT_CALLING:RUN_DEEPVARIANT:DEEPVARIANT (WTCHG_722292)'

Caused by:
  Failed to pull singularity image
  command: singularity pull  --name google-deepvariant-1.3.0.img.pulling.1664558111355 docker://google/deepvariant:1.3.0 > /dev/null
  status : 255
  message:
    FATAL:   While making image from oci registry: error fetching image to cache: failed to get checksum for docker://google/deepvariant:1.3.0: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp 44.205.64.79:443: i/o timeout

**Comment by maxulysse** [comment]
ok, then issue is to be fixed in tools.
@pontus suggested it probably has to do with the single/double quotes -> https://nextflow.slack.com/archives/C02TLU6K87P/p1664543901827989
I'll transfer this issue to tools then
